pkg/sql/schemachanger/eav/values.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package eav
+
+import "sync"
+
+// Values is used to store a select elements from a Tree.
+type Values map[Ordinal]Value
```

Comment:
This could also be a 
```
type Values struct {
    OrdinalSet
    values []Value
}
```
where `values` gets allocated to a capacity of 64.

I tried to make this change to benchmark it but it got annoying, because we then need to pass `Values` by reference. In any case you told me you'd considered doing it. As long as ordinals are capped at 64 it probably doesn't matter much either way. 

Anyway if you consider changing the implementation of `Values` at some point then you'd do well to wrap the map in a struct or something that will force you to pass by reference with `*Values`.